ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/DistributingExecutor.java

```diff
@@ -78,4 +98,38 @@ public DistributingExecutor(
           statement.getStatementText()), e);
     }
   }
+
+  private void validateSinkWritePermissions(
+      final CreateAsSelect createAsSelect,
+      final ServiceContext userContext,
+      final KsqlExecutionContext serverExecutionContext
+  ) {
+    final StatementDescriptor desc = StatementDescriptor.of(
+        createAsSelect,
+        serverExecutionContext.getMetaStore()
+    );
+
+    kafkaAccessValidator.validateWrite(desc.getSinkTopics(), userContext);
+
+    try {
+      kafkaAccessValidator.validateWrite(
+          desc.getSinkTopics(),
+          serverExecutionContext.getServiceContext()
+      );
+    } catch (final KsqlTopicAuthorizationException e) {
+      throw new KsqlServerException(String.format(
+          String.format(
+              "The KSQL service principal is not authorized to execute the command%nCaused by: %s",
+              e.getMessage()
+          )
+      ), e);
+    } catch (final Exception e) {
+      throw new KsqlServerException(String.format(
+          String.format(
+              "The KSQL service principal failed to validate the command%nCaused by: %s",
+              e.getMessage()
+          )
+      ), e);
+    }
```

Comment:
This is duplicate, but I didn't to mess with the `StatementPermissionsValidator` methods just for this error message. However, I plan to remove this by including the username in the auth error messages like: `KSQL server (User:ksql) is denied Write operation on topic(s): [t1]`. This way I just rely on the `KsqlAuthorizationException` message. I will do this in a follow-up PR.